### PR TITLE
Foreman Client el10 build support

### PR DESCRIPTION
**Pull request by pcreech**

### CODE REVIEW

File: comps/comps-foreman-client-el10.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  <!-- Meta information will go here eventually -->
+  <!--  </meta> -->
+  <group>
+    <id>foreman-client</id>
+    <name>Foreman Client</name>
+    <description>Foreman Client Packages</description>
+    <uservisible>true</uservisible>
+    <packagelist>
+      <packagereq type="default">foreman-client-release</packagereq>
+      <packagereq type="default">foreman_ygg_migration</packagereq>
```
[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:14]: @adamruzicka am I right here?

File: comps/comps-foreman-client-el10.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <packagereq type="default">foreman_ygg_worker</packagereq>
+      <packagereq type="default">katello-host-tools</packagereq>
+      <packagereq type="default">katello-host-tools-tracer</packagereq>
+      <packagereq type="default">katello-pull-transport-migrate</packagereq>
+      <packagereq type="default">python3-psutil</packagereq>
+      <packagereq type="default">rubygem-foreman_scap_client</packagereq>
+      <packagereq type="default">yggdrasil</packagereq>
```
[the remark this answers is not in the document]
Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:21]: Actually, it even exists in EL 9.5 so we might even be able to drop it there as well.

File: comps/comps-foreman-client-el10.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <description>Foreman Client Packages</description>
+    <uservisible>true</uservisible>
+    <packagelist>
+      <packagereq type="default">foreman-client-release</packagereq>
+      <packagereq type="default">foreman_ygg_migration</packagereq>
+      <packagereq type="default">foreman_ygg_worker</packagereq>
+      <packagereq type="default">katello-host-tools</packagereq>
+      <packagereq type="default">katello-host-tools-tracer</packagereq>
+      <packagereq type="default">katello-pull-transport-migrate</packagereq>
+      <packagereq type="default">python3-psutil</packagereq>
```
[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:19]: Also on el8 (we ship a newerr version) and el9 (el9 ships a newer version), so we could also drop on el9 at least

File: comps/comps-foreman-client-el10.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamruzicka) [reply comps/comps-foreman-client-el10.xml:14]: We won't need to migrate while on 10, but in theory, you could still do an in-place upgrade using leapp from rhel9 with older yggdrasil with a non-migrated config to rhel 10 and all the newer thing and then you'd still need it?

File: comps/comps-foreman-client-el10.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    <name>Foreman Client</name>
+    <description>Foreman Client Packages</description>
+    <uservisible>true</uservisible>
+    <packagelist>
+      <packagereq type="default">foreman-client-release</packagereq>
+      <packagereq type="default">foreman_ygg_migration</packagereq>
+      <packagereq type="default">foreman_ygg_worker</packagereq>
+      <packagereq type="default">katello-host-tools</packagereq>
+      <packagereq type="default">katello-host-tools-tracer</packagereq>
+      <packagereq type="default">katello-pull-transport-migrate</packagereq>
```
[the remark this answers is not in the document]
Comment (jeremylenz) [reply comps/comps-foreman-client-el10.xml:18]: We still document that the "correct" way to get pull mode running on an existing (registered) host is to install `katello-pull-transport-migrate`. It's confusingly named but works fine for that purpose.

https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#Configuring_a_Host_to_Use_the_Pull_Client_managing-hosts

[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:18]: TIL, thanks!

File: comps/comps-foreman-client-el10.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:14]: Is that something that leapp could/should address?

File: comps/comps-foreman-client-el10.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:21]: https://bugzilla.redhat.com/show_bug.cgi?id=2274334 mentions that it didn't make it into RHEL 9.5 so disregard my last comment.

File: comps/comps-foreman-client-el10.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:14]: in which way? migrating the config? could, certainly. but not sure it should.

but also, looking at https://github.com/oamg/leapp-repository/blob/main/repos/system_upgrade/common/files/upgrade_paths.json, I get the impression only 9.6+ will be supported for upgrades, so people will already have ygg 0.4

[the remark this answers is not in the document]
Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:14]: > in which way? migrating the config? could, certainly. but not sure it should.

Perhaps inhibit if < 0.4 is detected?

> but also, looking at https://github.com/oamg/leapp-repository/blob/main/repos/system_upgrade/common/files/upgrade_paths.json, I get the impression only 9.6+ will be supported for upgrades, so people will already have ygg 0.4

Does leapp guarantee all packages are fully updated before upgrading?

[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:14]: > Perhaps inhibit if < 0.4 is detected?

That's certainly something that you could suggest.

> Does leapp guarantee all packages are fully updated before upgrading?

No, it only checks that the release is correct. But you know what they say about systems with packages from different releases :)

[the remark this answers is not in the document]
Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:14]: I know that if you don't guarantee it, you're guaranteed to find a user with that problem. What's the best process around this? File a GitHub issue or straight to JIRA?

File: comps/comps-foreman-client-el10.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (pcreech) [reply comps/comps-foreman-client-el10.xml:18]: So... keep katello-pull-transport-migrate?

File: comps/comps-foreman-client-el10.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcreech) [reply comps/comps-foreman-client-el10.xml:14]: So, for the sake of the PR, i'm hearing "take it out, we can put it back later if needed"?

File: comps/comps-foreman-client-el10.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:18]: yes, keep it please

File: comps/comps-foreman-client-el10.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (evgeni) [reply comps/comps-foreman-client-el10.xml:14]: That's what Evgeni would do, putting things back in is easier to cleaning them up after they have existed.

File: package_manifest.yaml [hunk 5]
```diff
@@ -490,6 +495,7 @@ foreman_client_packages:
       el7:
         - "el7-foreman-client-{{ foreman_version }}-staging"
     repoclosure_lookaside_repos:
+      el10: "{{ hostvars['foreman-client-staging-repoclosure-el10']['repoclosure_lookaside_repos']['el10'] }}"
```
Comment (evgeni) [inline package_manifest.yaml:498]: You also need to adjust `repoclosure_target_repos` above ;)

File: comps/comps-foreman-client-el10.xml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  <!--  </meta> -->
+  <group>
+    <id>foreman-client</id>
+    <name>Foreman Client</name>
+    <description>Foreman Client Packages</description>
+    <uservisible>true</uservisible>
+    <packagelist>
+      <packagereq type="default">foreman-client-release</packagereq>
+      <packagereq type="default">foreman_ygg_migration</packagereq>
+      <packagereq type="default">foreman_ygg_worker</packagereq>
```
Comment (evgeni) [inline comps/comps-foreman-client-el10.xml:15]: @adamruzicka this fails to build on EL10:
https://download.copr.fedorainfracloud.org/results/evgeni/el10/centos-stream-10-x86_64/08343540-foreman_ygg_worker/builder-live.log.gz

Shouldn't block this PR, but something that will end up on your plate one day ;)

Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:15]: Copying discussion from Matrix: after https://github.com/theforeman/foreman-packaging/pull/11493 is merged we can investigate using the proper go macros to build. Those weren't available on EL7.

File: comps/comps-foreman-client-el10.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply comps/comps-foreman-client-el10.xml:19]: Let's move EL 9 removal to https://github.com/theforeman/foreman-packaging/pull/11561

**Comment by pcreech** [comment]
Updated :D

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
I have built katello-host-tools katello-pull-transport-migrate rubygem-foreman_scap_client foreman-release

**Pull request merged**